matrix_client/api.py

```diff
@@ -323,3 +338,21 @@ def _send(self, method, path, content=None, query_params={}, headers={}):
             )
 
         return response.json()
+
+    def media_upload(self,content,content_type):
```

Comment:
I'd prefer it if this was factored out along with `_send` and friends, as there is so much duplication between the two. The only notable difference AFAICS is the base endpoint URL (no reason why we can't make this an arg with `self.url` as the default) and whether or not it is JSON (which determines if we `json.dumps()` the `content`).